opaque_keys/__init__.py

```diff
@@ -329,6 +330,47 @@ def replace(self, **kwargs):
         existing_values.update(kwargs)
         return type(self)(**existing_values)
 
+    @classmethod
+    def field_strategy(cls, field):
+        """
+        A strategy to generate data of the right type for the fields of this OpaqueKey.
+
+        Arguments:
+            field: The name of the field to generate data for.
+        """
+        if field == 'deprecated':
+            return strategies.booleans()
+        else:
+            return strategies.text(min_size=1)
+
+    @classmethod
+    def key_strategy(cls):
+        """
+        A strategy to generate instances of this OpaqueKey class.
+        """
+        key_fields = {
+            field: cls.field_strategy(field)
+            for field in cls.KEY_FIELDS
+        }
+        return strategies.builds(
+            cls,
+            deprecated=cls.field_strategy('deprecated'),
+            **key_fields
+        )
+
+    @classmethod
+    def type_strategy(cls, blacklist=None):
+        """
+        A strategy to generate instances of this OpaqueKey KeyType.
+        """
+        if blacklist is None:
+            blacklist = ()
+        return strategies.one_of(*[
+            extension.plugin.key_strategy()
+            for extension in cls._drivers()
+            if not(issubclass(extension.plugin, blacklist))
+        ])
+
```

Comment:
I'd like these three methods clearly marked as being used in tests only. The opaque-keys module already requires a high cognitive load for mere mortals to understand.

As a general comment for all the new code below, I'd like to differentiate these methods and all the other hypothesis-based test methods so that anyone new to this module *or* trying to figure out something about its workings can ignore these methods if changing/understanding the tests isn't what they're doing at the moment.